consensus/safety-rules/src/safety_rules.rs

```diff
@@ -27,26 +28,40 @@ use libra_types::{
 /// set)
 pub struct SafetyRules {
     persistent_storage: PersistentSafetyStorage,
-    validator_signer: ValidatorSigner,
+    validator_signer: Option<ValidatorSigner>,
     validator_verifier: Option<ValidatorVerifier>,
 }
 
 impl SafetyRules {
     /// Constructs a new instance of SafetyRules with the given persistent storage and the
     /// consensus private keys
     /// @TODO replace this with an API that takes in a SafetyRulesConfig
+    /// @TODO change the constructor to fit Option<validator_signer> and address the circular
+    /// dependency with reconcile_key() during initializationg
     pub fn new(author: Author, persistent_storage: PersistentSafetyStorage) -> Self {
         let consensus_key = persistent_storage
```

Comment:
we don't know the public key at this point in time, we should not instantiate this.